cmake/f3dConfig.cmake.in

```diff
@@ -28,12 +41,15 @@ set(f3d_BINDINGS_JAVA "@F3D_BINDINGS_JAVA@")
 set(f3d_CONFIG_DIR "@f3d_config_dir@")
 set(f3d_PLUGINS_INSTALL_DIR "@F3D_PLUGINS_INSTALL_DIR@")
 
-# Provide a f3d_embed_file method for embedding image into source code
-include("${CMAKE_CURRENT_LIST_DIR}/f3dEmbed.cmake")
+# If no components specified, look for `library` component
+if(NOT f3d_FIND_COMPONENTS)
+  set(f3d_FIND_COMPONENTS library)
+endif()
+message(WARNING ${f3d_FIND_COMPONENTS})
 
-# Provide macros used to create a plugin
-include("${CMAKE_CURRENT_LIST_DIR}/f3dPlugin.cmake")
+foreach(component ${f3d_FIND_COMPONENTS})
+  # For requested component, execute its "config" script
+  include(${CMAKE_CURRENT_LIST_DIR}/${component}-config.cmake)
```

Comment:
What happens if the consumer asked for a component that does not exist?

Reply:
I've implemented something, looks ok to me but let me know